### PR TITLE
New package: libspotify-12.1.51

### PR DESCRIPTION
**Pull request by Pizza-Boy**

### CODE REVIEW

File: srcpkgs/libspotify/template [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#Template for the 'libspotify' package
+pkgname=libspotify
+version=12.1.51
+revision=1
+depends="glibc"
```
Comment (Duncaen) [inline srcpkgs/libspotify/template]: This is not needed

File: srcpkgs/libspotify/template [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+short_desc="Spotify library for building your own streaming apps"
+maintainer="noah <nsawyer1993@gmail.com>"
+only_for_arches="x86_64 i686 armv6 armv7"
```
Comment (Duncaen) [inline srcpkgs/libspotify/template]: type `only_for_arches` > `only_for_archs`

File: srcpkgs/libspotify/template [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+post_install() {
+	#Fix the pkgconfig file
+	sed -e s:PKG_PREFIX:/usr:g \
+		< lib/pkgconfig/libspotify.pc \
+		> ${DESTDIR}/usr/lib/pkgconfig/libspotify.pc
+	#Install our licenses
+	mkdir ${DESTDIR}/usr/share
+	mkdir ${DESTDIR}/usr/share/doc
+	mkdir ${DESTDIR}/usr/share/doc/libspotify
```
Comment (Duncaen) [inline srcpkgs/libspotify/template]: you can replace those three `mkdir` lines with `vmkdir usr/share/libspotify`

**Comment by Duncaen** [comment]
The pkgconfig and unversioned shared object should be part of a `libspotify-devel` sub package.

**Comment by Pizza-Boy** [comment]
I should have corrected most of your comments, the only thing I wasn't sure about was the unversioned shared object you were referring to.

**Comment by Pizza-Boy** [comment]
Adding the libspotify-devel package has caused this error: "PKGDESTDIR is empty and build_style != meta". I'll amend a new commit with some changes to the -devel package that add the unspecified shared object to the -devel package regardless.

File: srcpkgs/libspotify/template [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+libspotify-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/lib/*.so
+		#Fix the pkgconfig file
+		sed -i s:PKG_PREFIX:/usr:g lib/pkgconfig/libspotify.pc
```
Comment (Vaelatern) [inline srcpkgs/libspotify/template]: This should not be done in a subpackage. This fix should be probably made in `post_install()`.

File: srcpkgs/libspotify/template [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+do_install() {
+	make prefix=${DESTDIR}/usr install
+}
+
+post_install() {
+	#Install our licenses
+	vmkdir usr/share/doc/libspotify
+	install -Dm644 LICENSE licenses.xhtml ${DESTDIR}/usr/share/doc/libspotify
```
Comment (Vaelatern) [inline srcpkgs/libspotify/template]: Line not needed. `vlicense licenses.xhtml` with the below line (`vlicense LICENSE`) will do just fine. Similarly, the `vmkdir` line is not necessary.

File: srcpkgs/libspotify/template [hunk 6]
```diff
@@ -0,0 +1,54 @@
+#Template for the 'libspotify' package
```
Comment (Vaelatern) [inline srcpkgs/libspotify/template]: Convention is `# Template file for 'libspotify'`

**Comment by Gottox** [comment]
@Duncaen @Vaelatern: merge if you think it's okay.

**Comment by Pizza-Boy** [comment]
All of the suggested changes should have been added.

**Comment by Vaelatern** [comment]
We have a `-devel` package here, so I think we need a change to `common/shlibs`, yes?

**Comment by Pizza-Boy** [comment]
Is there any organization to the `shlibs` file or should I just throw that line in at the bottom?

**Comment by Gottox** [comment]
There's no special sorting. Just put it at the end.

**Comment by Pizza-Boy** [comment]
Should be pushing the new file now.

File: srcpkgs/libspotify/template [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	#Install our licenses
+	vlicense licenses.xhtml
+	vlicense LICENSE
+	#Fix the pkgconfig file
+	sed -i s:PKG_PREFIX:/usr:g lib/pkgconfig/libspotify.pc
+}
+
+libspotify-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}"
```
Comment (Vaelatern) [inline srcpkgs/libspotify/template]: `depends="${sourcepkg}>=${version}_${revision}"`

File: srcpkgs/libspotify-devel [hunk 8]
```diff
@@ -0,0 +1 @@
+libspotify/
```
Comment (Vaelatern) [inline srcpkgs/libspotify-devel]: Minor quibble. When you make the symlink, please don't include a trailing slash.

**Comment by Vaelatern** [comment]
Please rename the commit simply `New package: libspotify-12.1.51`

**Comment by Pizza-Boy** [comment]
done done and done

**Comment by Vaelatern** [comment]
Renaming the commit is not renaming the issue ;)

**Comment by Pizza-Boy** [comment]
To think it'd be so simple. Fixed.

File: srcpkgs/libspotify-devel/template [hunk 9]
```diff
@@ -0,0 +1 @@
+/home/noah/builds/void-packages/srcpkgs/libspotify/template
```
Comment (Vaelatern) [inline srcpkgs/libspotify-devel/template]: `cd srcpkgs`
`ln -s libspotify libspotify-devel`

Comment (Vaelatern) [inline srcpkgs/libspotify-devel/template]: Specifically the symlink should be directory to directory.

**Comment by Pizza-Boy** [comment]
hah, sorry, I'm bad at these `ln` commands.

**Comment by Vaelatern** [comment]
And now, to fix the conflicts, please rebase on top of the current master.

**Comment by Pizza-Boy** [comment]
I'm a bit new to git, after I rebase with `git rebase master` do I just push the commit again?

**Comment by Vaelatern** [comment]
Current master means your master branch must also be up to date with the upstream version of master branch, possibly with (if you have this repository named "upstream") `git checkout master && git pull upstream master`, then yes, rebase against master, then yes, force push the commit again. Your rebase command should only show you the one commit, otherwise you may have done something wrong.

**Comment by Vaelatern** [comment]
Try a `git rebase upstream/master` and push that again. Might iron out the merge commit.

**Comment by Pizza-Boy** [comment]
Alright, rebased it and pushed from the `master` branch, which I think should fix any conflicts/problems.

**Comment by Vaelatern** [comment]
Please try to get rid of the merge commit. Checking out the topic branch (`libspotify-branch`) and rebasing on `upstream/master` then pushing from that branch will likely iron it out.

**Comment by Pizza-Boy** [comment]
Alright checked out the current branch, rebased, and pushed, but it doesn't seem to want to clear the merge. Looks like that cleared everything up. 

Sorry about messing that up a bit!

File: srcpkgs/libspotify/template [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+	#Install our licenses
+	vlicense licenses.xhtml
+	vlicense LICENSE
+	#Fix the pkgconfig file
+	sed -i s:PKG_PREFIX:/usr:g lib/pkgconfig/libspotify.pc
+}
+
+libspotify-devel_package() {
+	short_desc+=" - development files"
+	depends="${sourcepkg}-${version}_${revision}"
```
Comment (Vaelatern) [inline srcpkgs/libspotify/template]: Ah, sorry, it should be a `>=` not a `-` in this line.

**Comment by Pizza-Boy** [comment]
It's all good. Fixed.

**Pull request merged**